io/directory.go

```diff
@@ -198,17 +198,28 @@ func (d *BasicDirectory) SetCidBuilder(builder cid.Builder) {
 // AddChild implements the `Directory` interface. It adds (or replaces)
 // a link to the given `node` under `name`.
 func (d *BasicDirectory) AddChild(ctx context.Context, name string, node ipld.Node) error {
+	link, err := ipld.MakeLink(node)
+	if err != nil {
+		return err
+	}
+
+	return d.addLinkChild(ctx, name, link)
+}
+
+// addLinkChild adds the link as an entry to this directory under the given
+// name. Plumbing function for the AddChild API.
+func (d *BasicDirectory) addLinkChild(ctx context.Context, name string, link *ipld.Link) error {
```

Comment:
nit: the name seems redundant given that `ipld.Link` already includes a name.

(I mean, ideally we could just use a CID... but we need the size because dagpb).

Reply:
My first version was actually just the link argument with the name packed inside but looking at how we manipulate links in [go-merkledag](https://github.com/ipfs/go-merkledag/blob/cbf224ca558ade3b90291698990c95136613faa2/node.go#L97-L123) it seemed like the idiomatic way to go was to maniuplate that separately. Even internally here in [go-unixfs](https://github.com/ipfs/go-unixfs/blob/ffa25dfdbad590760074ae89e5e5f5229fa394d0/hamt/hamt.go#L326-L327) we handle the name separately.

I totally agree with you and the current way seems a bit counter-intuitive to me so happy to revert to my previous version.

Reply:
No, leave it. Ideally we'd just get rid of that type entirely but that's work for another day.